numbast/src/numbast/static/renderer.py

```diff
@@ -0,0 +1,46 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+class BaseRenderer:
+    Prefix = """
+from pynvjitlink.patch import patch_numba_linker
+patch_numba_linker()
+"""
+
+    Imports: set[str] = set()
+    """Empty set to be filled later. One element stands for one line of import."""
+
+    MemoryShimWriterTemplate = """
+c_ext_shim_source = CUSource(\"""{shim_funcs}\""")
+"""
+
+    includes_template = "#include <{header_path}>"
+    """Template for including a header file."""
+
+    Includes: set[str] = set()
+    """includes to add in c extension shims."""
```

Comment:
I've had doubts whether `Includes` and `Imports` should be a class variable. There are arguments on both sides: since eventually we are creating 1 file that contains all the bindings, we need a consolidated place to track the imports and reduce duplication. On the other hand, each renderer depends on different things. It could become harder to debug when multiple renderers generate at the same time. I'd like to postpone the decision now and see what it becomes as the developement continues.

Reply:
I understand your reasoning here. I think postponing the decision makes sense.